src/main/scala/org/bitcoins/core/serializers/transaction/RawWitnessTransactionParser.scala

```diff
@@ -0,0 +1,52 @@
+package org.bitcoins.core.serializers.transaction
+
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.transaction.{TransactionWitness, WitnessTransaction}
+import org.bitcoins.core.serializers.RawBitcoinSerializer
+import org.bitcoins.core.util.BitcoinSUtil
+
+/**
+  * Created by chris on 11/21/16.
+  */
+trait RawWitnessTransactionParser extends RawBitcoinSerializer[WitnessTransaction] {
+
+  def read(bytes: List[Byte]) : WitnessTransaction = {
+    val versionBytes = bytes.take(4)
+    val version = UInt32(versionBytes.reverse)
+    val marker = bytes(4).toChar
```

Comment:
This seems pretty hacky wrt to `marker` and `flag`, find where they check these two requirements in bitcoin core and document why these invariants need to hold true